formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Encoding.kt

```diff
@@ -392,20 +656,67 @@ internal class CborDecoder(private val input: ByteArrayInput) {
             input.readExactNBytes(strLen)
         }
 
-    private fun skipOverTags() {
+    private fun processTags(tags: ULongArray?): ULongArray? {
+        val collectedTags = mutableListOf<ULong>()
         while ((curByte and 0b111_00000) == HEADER_TAG) {
-            readNumber() // This is the tag number
+            val readTag = readNumber().toULong() // This is the tag number
+            collectedTags += readTag
             readByte()
         }
+
+        return (if (collectedTags.isEmpty()) null else collectedTags.toULongArray()).also { collected ->
+            tags?.let {
+                if (!it.contentEquals(collected)) throw CborDecodingException(
```

Comment:
Note that `contentEquals` has both nullable receiver and parameter. It results in behavior like this: `intArrayOf(1,2,3).contentEquals(null) -> false`. I'm not sure it is what you intended to have here.

Reply:
We only want to compare if tags are actually set. Otherwise, we don't care.

I'll also add that comment to the code.